pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -196,8 +196,13 @@ func (l *PodInterface) PlugPhase1(vmi *v1.VirtualMachineInstance, iface *v1.Inte
 			return err
 		}
 
+		queueNumber := uint32(0)
 		isMultiqueue := (vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue != nil) && (*vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue)
-		if err := driver.preparePodNetworkInterfaces(isMultiqueue, pid); err != nil {
+		if isMultiqueue {
+			cpuTopology := api.GetCPUTopology(vmi)
+			queueNumber = api.CalculateRequestedVCPUs(cpuTopology)
```

Comment:
I don't think that the tap device is actually interested in the number of queues, except that we can add the multiqueue flag, but this "duplicates" the network queue calculation in a sense. We use the same way inside the converter to detect it, but there it was one isolated location.

I would feel better if we can introduce a more meaningful function, e.g. `CalculateNetworkQueueNumber` and use it inside the converter and here. This way we encapsulate the knowledge better and the risk to calculate different queue numbers of times is reduced.

Reply:
Since the converter requires both the CPUTopology struct and the number of queues, I've bundled those 2 into a single function.

Hope that's OK.